packages/manager/apps/pci-databases-analytics/src/pages/services/[serviceId]/users/_components/formUser/AclsSelect.component.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useEffect, useMemo, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+import { z } from 'zod';
+import { SubmitHandler } from 'react-hook-form';
+import { InfoIcon, MinusCircle, PlusCircle } from 'lucide-react';
+import { UserAcl } from '@/types/cloud/project/database/opensearch';
+import {
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@/components/ui/form';
+import { useAclsSelectForm } from './useAclsSelectForm.hook';
+import { Input } from '@/components/ui/input';
+import { Button } from '@/components/ui/button';
+import { ScrollArea } from '@/components/ui/scroll-area';
+import { Badge } from '@/components/ui/badge';
+import {
+  Select,
+  SelectContent,
+  SelectGroup,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from '@/components/ui/select';
+import { USER_CONFIG } from './user.constants';
+import { Alert, AlertDescription } from '@/components/ui/alert';
+
+interface AclsSelectProps {
+  value?: UserAcl[];
+  onChange?: (newAcls: UserAcl[]) => void;
+}
+
+const AclsSelect = React.forwardRef<HTMLInputElement, AclsSelectProps>(
+  ({ value, onChange }, ref) => {
+    const { t } = useTranslation(
+      'pci-databases-analytics/services/service/users',
+    );
+    const patternInputRef = useRef<HTMLInputElement>(null);
+    const addAclBtnRef = useRef<HTMLButtonElement>(null);
+    const scrollListRef = useRef<HTMLUListElement>(null);
+    const { form, schema } = useAclsSelectForm({
+      existingAcls: value,
+    });
+    type ValidationSchema = z.infer<typeof schema>;
+
+    // Scroll to the bottom of the list when a new acl is added
+    useEffect(() => {
+      const scrollContainer = scrollListRef.current;
+      if (scrollContainer) {
+        const { lastChild } = scrollContainer;
+        if (lastChild instanceof Element) {
+          lastChild.scrollIntoView({ behavior: 'smooth' });
+        } else {
+          scrollContainer.scrollTop = scrollContainer.scrollHeight;
+        }
+      }
+    }, [value, scrollListRef]);
+
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+      if (event.key === 'Enter') {
+        addAclBtnRef.current?.click();
+        patternInputRef.current?.focus();
+        event.preventDefault();
+      }
+    };
+    const handleAddAcl: SubmitHandler<ValidationSchema> = (formValues) => {
+      onChange([
+        ...value,
+        {
+          pattern: formValues.pattern,
+          permission: formValues.permission,
+        },
+      ]);
+      form.reset();
+      patternInputRef.current?.focus();
+    };
+    const handleRemoveAcl = (index: number) => {
+      const updatedAcls = value.filter((_, i) => i !== index);
+      onChange(updatedAcls);
+    };
```

Comment:
Could a bit more elegant by adding some currification and removing the uncesserary lambda function in JSX event: 

```suggestion
    const handleRemoveAcl = (index: number) => () => {
      const updatedAcls = value.filter((_, i) => i !== index);
      onChange(updatedAcls);
    };
```